### PR TITLE
fix: refactor PrimMST and fix bug in PriorityQueue

### PR DESCRIPTION
**Pull request by paulinegarelli**
Fixes: #1298 

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [ ] Add an algorithm?
- [x] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

Refactored `PrimMST` by moving `PriorityQueue` into its own algorithm file in `Data-Structures\Heap\KeyPriorityQueue.js`
Added tests for `PrimMST` and `KeyPriorityQueue`
Fixed bug in `PriorityQueue` (method `_shiftDown`)

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Data-Structures/Heap/KeyPriorityQueue.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+function getChildrenPosition (position) {
+  // Get the children nodes of the current node
+  return [2 * position + 1, 2 * position + 2]
+}
+
+class KeyPriorityQueue {
+  // Priority Queue class using Minimum Binary Heap
+  constructor () {
+    this._heap = []
+    this.keys = {}
```
Comment (appgurueu) [inline Data-Structures/Heap/KeyPriorityQueue.js:35]: Why not use a ES6 `Map` here?

File: Data-Structures/Heap/KeyPriorityQueue.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+ *
+ * More information and visuals on Binary Heaps can be found here: https://www.geeksforgeeks.org/binary-heap/
+ */
+
+// Priority Queue Helper functions
+function getParentPosition (position) {
+  // Get the parent node of the current node
+  return Math.floor((position - 1) / 2)
+}
+function getChildrenPosition (position) {
```
Comment (appgurueu) [inline Data-Structures/Heap/KeyPriorityQueue.js:26]: Same here. Also please rename to `getChildrenPosition`**`s`**.

File: Data-Structures/Heap/test/KeyPriorityQueue.test.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
+import { KeyPriorityQueue } from '../KeyPriorityQueue.js'
+
```
Comment (appgurueu) [inline Data-Structures/Heap/test/KeyPriorityQueue.test.js:2]: You should wrap your tests in one big `describe("Key Priority Queue", () => { ... })`.

File: Data-Structures/Heap/KeyPriorityQueue.js (hunk 1 above)
Comment (paulinegarelli) [reply Data-Structures/Heap/KeyPriorityQueue.js]: Indeed it helps clarify the code.
I made the change.

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**